model/jpa/src/main/java/org/keycloak/connections/jpa/DefaultJpaConnectionProviderFactory.java

```diff
@@ -91,8 +98,24 @@ public JpaConnectionProvider create(KeycloakSession session) {
             em = emf.createEntityManager(SynchronizationType.SYNCHRONIZED);
         }
         em = PersistenceExceptionConverter.create(session, em);
-        if (!jtaEnabled) session.getTransactionManager().enlist(new JpaKeycloakTransaction(em));
-        return new DefaultJpaConnectionProvider(em);
+        if (!jtaEnabled) {
+            session.getTransactionManager().enlist(new JpaKeycloakTransaction(em));
+        }
+        return em;
+    }
+
+    private void addSpecificNamedQueries(KeycloakSession session, Connection connection) {
```

Comment:
We should avoid whenever possible performing blocking/costly operations during startup.

Also, would be possible to just use the connection metadata to obtain the database type rather than querying the liquibase provider? Asking because Liquibase operations are usually costly.

Reply:
+1. Should be done in augmentation. I wonder if this is / should be done for the whole quarkus/runtime/src/main/java/org/keycloak/connections/jpa/QuarkusJpaConnectionProviderFactory.java?

Reply:
Yeah ... There are other parts of the code that we can move to build time. For instance, anything that touches upgrade/migration should happen when re-augmenting the server. That could be done once (or manually triggered by the user) and just leave the very necessary bits to the runtime.